### ISSUE TITLE
VCAP_SERVICES properties not found with version 2.13.0

### ISSUE THREAD

**Issue by adamandre79** [SAP/cloud-security-xsuaa-integration#955]
Hello,

I have upgraded the version off `xsuaa-spring-boot-starter` from 2.12.0 to 2.13.0 in my project.
In addition I have created a bean for the RestTemplate.

After deploying the app to a cloud foundry space and try to access it with a REST request I always get the error in the logs: `Failed to authenticate since the JWT was invalid`. 
I debugged the application and found, that the class `XsuaaServiceConfigurationDefault` has no properties filled and because the property `uaadomain` is empty the JWT can't be validated.

In the same change we also upgraded our project to spring boot 2.7.0. Can this be the reason why the properties are not filled.

I checked in the space the VCAP_SERVICES and the env variable is filled and one xsuaa section is included with the needed properties, but somehow the properties are not found.

path the  `uaadomain`: VCAP_SERVICES -> "xsuaa" -> "credentials" -> "uaadomain"

I know I could implement an own `XsuaaCredentials` but would like to understand why it is not working and what could be the reason.
At the moemnt we have not implemented the `XsuaaCredentials` on our own.

Regards
André

**Comment by adamandre79** [comment]
Hello,

I think I found the issue but need to verify. If it works as expected we had a wrong bean definition.
I will check this, and if the error is solved I will update and close the issue.

Regards
André

**Comment by adamandre79** [comment]
Hi,

the problem was that with version 2.12.0 of the library `xsuaa-spring-boot-starter` we need to create a bean of type `XsuaaServiceConfiguration`.
We created it with:

```
    @Bean
    XsuaaServiceConfiguration xsuaaConfiguration() {
        return new XsuaaServiceConfigurationDefault();
    }
```

Now after upgrading to version 2.13.0 this bean was not working and has caused all the trouble we had.
I have not found an migration guide from 2.12.0 to 2.13.0 and so we were not aware that this bean creation causes all the trouble.
I only found it by debugging the library.

For us the problem is solved but I really sure also other teams will have this problem which causes a lot of waist time to investigate.

Regards
André

**Comment by liga-oz** [comment]
Hi @adamandre79,

could you elaborate more on how you fixed the problem? With the 2.13.0 the major change was that we started consuming another library that reads the environment variables, but from the testing done, there was no issues found and everything worked as before. I wonder whether this is something to do with the environment library or the spring boot version 2.7.x

What environment you're app is deployed on?

Kind Regards,
Liga

**Comment by adamandre79** [comment]
Hi,

we are deploying on cloud foundry.

Regards
André
